app/Repositories/Subjects/SubjectRepon.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+namespace App\Repositories\Subjects;
+
+use App\Models\Subject;
+
+use App\Repositories\BaseRepository;
+use App\Repositories\Subjects\SubjectInterface;
+
+/**
+ * Class EquipmentRepository
+ * @package App\Repositories\Equipment
+ */
+class SubjectRepon extends BaseRepository implements SubjectInterface
+{
+    //protected $subject;
+
+    public function __construct(Subject $subject)
+    {
+        $this->model = $subject;
+    }
+
+    public function getListSubject($request)
+    {
+       $list = $this->model;
+       if (!empty($request['name'])) {
+        return [
+            'success' => true,
+            'data' => $list->where('name', 'like', '%' . $request['name'] . '%')->paginate(5)
+        ];
+    }
+    return [
+        'success' => true,
+        'data' =>$list->paginate(5)
+    ];
+
+    }
+
+    public function deleteSubject($id)
+    {  
+        $subject = Subject::find($id);
```

Comment:
public function __construct(Subject $subject)
    {
        $this->model = $subject;
    }
em map model subject ở bên  trên rồi thì ở dưới thì cần gọi $this->model là sẽ tương đương với Subject:: rồi nhé